server/src-rsr/initialise.sql

```diff
@@ -1,5 +1,14 @@
+/* We define our own uuid generator function that uses gen_random_uuid() underneath.
+Since the column default is not directly referencing gen_random_uuid(),
+it prevents the column default to be dropped when pgcrypto or public schema is dropped unwittingly.
+
+See https://github.com/hasura/graphql-engine/issues/4217
+*/
+CREATE FUNCTION hdb_catalog.gen_hasura_uuid() RETURNS uuid AS
+'select gen_random_uuid()' LANGUAGE SQL STABLE;
```

Comment:
Ideally it has to be `select <schema-where-pgcrypto-is-installed>.gen_random_uuid()`. We don't do it on master but it'll be great if it is added as a TODO comment in the code.